### ISSUE TITLE
feat: NomicEmbeddingProvider — local ONNX EmbeddingService for zero-latency query embedding

### ISSUE THREAD

**Issue by sparck75** [memoryrelay/openclaw-plugin#111]
## Context

PR #110 adds `ApiEmbeddingService` (server-side embedding via POST /v1/embed, ~50-200ms latency). This issue tracks the complementary local implementation for users who want zero-network-latency vector search.

## Why local?

- **Latency**: ~5ms vs ~50-200ms for API round-trip
- **Offline support**: works without a network connection
- **Privacy**: no query text leaves the device

## Prior art

The MemoryRelay API already has a production Nomic ONNX provider merged in PR #372:
- `app/services/embedding/nomic_provider.py`
- Uses `onnxruntime` + `huggingface_hub` + `tokenizers`
- Downloads `nomic-ai/nomic-embed-text-v1.5` (~131 MB quantized ONNX) on first use
- Applies asymmetric task prefixes (`search_query:` / `search_document:`)
- L2-normalized output (cosine similarity = dot product)

## Implementation plan

Port the Python `NomicEmbeddingProvider` to TypeScript using `onnxruntime-node`:

```typescript
// src/cache/nomic-embedding-service.ts
import type { EmbeddingService } from "../pipelines/types.js";

export class NomicEmbeddingService implements EmbeddingService {
  private session: any = null;
  private tokenizer: any = null;
  
  async generateQuery(text: string): Promise<Float32Array> {
    await this.ensureLoaded();
    return this._embed(`search_query: ${text}`);
  }
  
  // ... lazy model loading via @huggingface/hub download
  // ... tokenize → ONNX inference → mean pool → L2 normalize
}
```

**Dependencies (optional — only loaded when provider=nomic):**
- `onnxruntime-node` (~50 MB)
- `@huggingface/hub` (model download)
- `@xenova/transformers` tokenizer (or port the Python tokenizer logic)

## Trade-offs

| | ApiEmbeddingService | NomicEmbeddingService |
|---|---|---|
| Install size | +0 MB | +50 MB (onnxruntime) + 131 MB model |
| First-call latency | ~50-200 ms (network) | ~2-5s (model download, once) |
| Per-call latency | ~50-200 ms | ~5 ms |
| Offline support | ❌ | ✅ |
| Privacy | Sends text to API | Local only |

## Activation

```jsonc
{
  "localCache": {
    "vectorSearch": {
      "enabled": true,
      "provider": "nomic"  // vs "api" (default when enabled)
    }
  }
}
```

In `index.ts`:
```typescript
const embeddingService = pluginConfig.vectorSearch?.enabled
  ? pluginConfig.vectorSearch.provider === "nomic"
    ? new NomicEmbeddingService()
    : new ApiEmbeddingService(client)  // default
  : undefined;
```

## Acceptance criteria

- [ ] `NomicEmbeddingService` implements `EmbeddingService` interface
- [ ] Lazy model download on first call (~131 MB, cached)
- [ ] `generate_query()` applies `search_query:` prefix
- [ ] Output is L2-normalized Float32Array (768 dims)
- [ ] `index.ts` selects provider based on `vectorSearch.provider` config
- [ ] Does NOT break existing behaviour when `vectorSearch.enabled = false`

**Comment by sparck75** [comment]
## Implementation Plan

### Overview
Add a `NomicEmbeddingService` that runs the Nomic-embed-text-v1.5 ONNX model locally via `onnxruntime-node`. Selected when `localCache.vectorSearch.provider = "nomic"`.

### Files to create/modify

**1. `src/cache/nomic-embedding-service.ts` (new, ~150 lines)**

```typescript
import type { EmbeddingService } from "../pipelines/types.js";

export class NomicEmbeddingService implements EmbeddingService {
  private session: any = null;
  private tokenizer: any = null;
  private readonly cacheDir: string;

  constructor(cacheDir: string) {
    this.cacheDir = cacheDir;  // e.g. ~/.openclaw/models/
  }

  async generateQuery(text: string): Promise<Float32Array> {
    await this.ensureLoaded();
    return this._embed(`search_query: ${text}`);
  }

  private async ensureLoaded(): Promise<void> {
    if (this.session) return;
    const { download, run } = await this.loadDeps();
    const modelPath = path.join(this.cacheDir, "nomic-embed-text-v1.5.Q8_0.onnx");
    if (!fs.existsSync(modelPath)) {
      await download(modelPath);  // ~131 MB, once
    }
    this.session = await run(modelPath);
    this.tokenizer = await loadTokenizer();  // from @xenova/transformers or tokenizers
  }

  private async _embed(text: string): Promise<Float32Array> {
    const tokens = await this.tokenizer.encode(text);
    const inputIds = new BigInt64Array(tokens.ids.map(BigInt));
    const attentionMask = new BigInt64Array(tokens.attention_mask.map(BigInt));
    const result = await this.session.run({
      input_ids: { data: inputIds, dims: [1, inputIds.length] },
      attention_mask: { data: attentionMask, dims: [1, attentionMask.length] },
    });
    const embeddings = result.last_hidden_state.data;
    // Mean pool over sequence length
    const dim = 768;
    const seqLen = inputIds.length;
    const pooled = new Float32Array(dim);
    for (let d = 0; d < dim; d++) {
      let sum = 0;
      for (let s = 0; s < seqLen; s++) sum += embeddings[s * dim + d];
      pooled[d] = sum / seqLen;
    }
    // L2 normalize
    const norm = Math.sqrt(pooled.reduce((s, v) => s + v * v, 0));
    return pooled.map(v => v / norm);
  }
}
```

**2. `package.json` — add optional peer deps**

```json
"peerDependencies": {
  "onnxruntime-node": ">=1.17.0",
  "@xenova/transformers": ">=2.17.0"
},
"peerDependenciesMeta": {
  "onnxruntime-node": { "optional": true },
  "@xenova/transformers": { "optional": true }
}
```

**3. `index.ts` — wire provider selection**

```typescript
// Replace current ApiEmbeddingService instantiation:
const embeddingService: EmbeddingService | undefined =
  !pluginConfig.vectorSearch?.enabled
    ? undefined
    : pluginConfig.vectorSearch.provider === "nomic"
      ? new NomicEmbeddingService(path.join(os.homedir(), ".openclaw", "models"))
      : new ApiEmbeddingService(client);  // default: API-backed
```

**4. `openclaw.plugin.json` — expose config option**

```json
"provider": {
  "type": "string",
  "enum": ["api", "nomic"],
  "default": "api",
  "description": "Embedding provider: 'api' (server-side via POST /v1/embed, default) or 'nomic' (local ONNX, ~131MB download on first use)"
}
```

### Trade-offs

| | api (default) | nomic (local) |
|---|---|---|
| Install size | +0 MB | +50 MB (onnxruntime) + 131 MB model |
| First-call latency | ~50-200 ms | ~2-5s (model load, once) |
| Per-call latency | ~50-200 ms | ~5 ms |
| Network | Requires API access | Fully offline |
| Privacy | Query text sent to API | Local only |

### Complexity: Medium (2-3 days)
Dependencies: onnxruntime-node, @xenova/transformers (both optional)
Risk: ONNX model download UX needs care (progress indicator, first-use warning)
